bundler/lib/bundler/ruby_dsl.rb

```diff
@@ -11,7 +11,12 @@ def ruby(*ruby_version)
 
       if options[:file]
         raise GemfileError, "Cannot specify version when using the file option" if ruby_version.any?
-        ruby_version << Bundler.read_file(Bundler.root.join(options[:file])).strip
+        file_content = Bundler.read_file(Bundler.root.join(options[:file]))
```

Comment:
Perhaps it is not intended to support the variability of `.ruby-version`, but this doesn't currently support the standard `ruby-3.1.2` syntax that is an option for `.ruby-version`.